### PR TITLE
arch: migrate lsp find_refs to new sema model

### PR DESCRIPTION
**Pull request by XiaoK29**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [X] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm/tools/src/LSP/src/find_refs.rs
kclvm/tools/src/LSP/src/request.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [X] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [X] Other

### CODE REVIEW

File: kclvm/tools/src/LSP/src/find_refs.rs [hunk 1]
```diff
@@ -13,46 +14,37 @@ use std::collections::HashMap;
...
     vfs: Option<Arc<RwLock<Vfs>>>,
     logger: F,
+    gs: &GlobalState,
 ) -> Result<Vec<Location>, String> {
-    // find the definition at the position
-    let def = match program.pos_to_stmt(kcl_pos) {
-        Some(node) => match node.node {
-            Stmt::Import(_) => None,
-            _ => find_def(node.clone(), kcl_pos, prog_scope),
+   let def = find_def_with_gs(kcl_pos, &gs, true);
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/find_refs.rs:26]: Could U pls format this code to pass the CI check?

Comment (XiaoK29) [reply kclvm/tools/src/LSP/src/find_refs.rs]: ok

Comment (Peefy) [reply kclvm/tools/src/LSP/src/find_refs.rs]: > ok

The CI still fails. 
image

**Review by Peefy — APPROVED** [review]
LGTM

File: kclvm/tools/src/LSP/src/find_refs.rs [hunk 2]
```diff
@@ -13,46 +14,41 @@ use std::collections::HashMap;
 use std::sync::Arc;
 
 pub(crate) fn find_refs<F: Fn(String) -> Result<(), anyhow::Error>>(
-    program: &Program,
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/find_refs.rs:16]: Unit test failed in CI: https://github.com/kcl-lang/kcl/actions/runs/6939682497/job/18877398025?pr=899

**Pull request merged**